text/0000-cargo-tags.md

```diff
@@ -0,0 +1,126 @@
+2. Is not a crate tag. (These use the crate for their metadata instead.)
+3. Has at least five "noteworthy" crates from separate authors. (Intentionally vague to allow unpopular but important tags to gain metadata.)
+4. Serve a purpose that is not covered by an existing tag. (For these cases, subtags are more appropriate. A good metric for this would be whether the five noteworthy crates share a second tag in common that itself is.)
+5. Could feasibly have at least one subtag that is noteworthy. ("Feasibly" is also left highly subject to interpretation. The idea here is to ensure that these are not best replaced with subtags.)
+
+For determining the noteworthiness of subtags, the conditions are slightly altered:
+
+* They only have to serve a purpose that isn't covered by other subtags, instead of any tag.
+* They don't even feasibly need to have noteworthy subtags, since nested subtags aren't allowed.
+
+When determining noteworthiness, similar tags can be taken into account. For example, if `math`, `maths`, and `mathematics` are used as tags, but the community hasn't yet decided upon a canonical tag, one can be chosen arbitrarily and metadata can be added to it instead.
+
+Metadata can take one of the following forms:
+
+* A short description can be added to the tag.
+* A list of similar tags can be added, grouping together crates under all these tags in the same list.
+
+Additionally, tags that are seen as noteworthy are allowed to be shown on the "popular tags" list on the crates.io home page, or displayed prominently above other tags on crate pages.
+
+These guidelines are meant to provide a *minimum* requirement to add a noteworthy tag, but meeting these requirements does not ensure that a tag is added. These tags may still be vetoed at the discretion of the crates.io team, and this may also result in changes to the policy.
+
+Since this is a big cultural change to the way crates are presented, it's going to take a bit of experimentation and time to fully refine these guidelines. This is why the crates.io team should be allowed to change these guidelines as they see fit without a full RFC, since ultimately, they're the ones in charge of actually responding to changes to tag metadata and they can only do what they have the capacity to do.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+The biggest drawback is that this can potentially add substantial extra work for the crates.io team, since there could potentially be several more tags with metadata than existing categories.
+
+However, the requirements here aren't substantially different than those expected of users proposing categories before, just more clarified. Arguably, the onus is more on the users proposing tags than the crates.io members accepting metadata changes, since it's substantially more difficult to make a case than to accept or reject one.
+
```

Comment:
Search seems like an important part of usability, but it's not mentioned in this RFC at all.

```suggestion

Another drawback is the proliferation of tags with similar or misspelt names. This makes searching for tags more difficult than categories. For example, if the tags `math`, `maths`, and `mathematics` are all in common use, a string-based search for `maths` (a common spelling in many countries) won't find the other maths-related tags. This reduces search usability, particularly for users whose English dialects use variant spellings. (And improving search usability complicates the search implementation.) It could also lead to a proliferation of tags with identical meanings, but variant spellings. However, all these drawbacks already apply to keywords.
```

Reply:
we could potentially keep a list of aliases on the backend for some of the "noteworthy" tags mentioned above. once we have chosen a noteworthy tag I assume most crate authors would probably adjust their usage of the tags to appear in the list of related crates.

Reply:
This is kind of what I tried to clarify in the RFC, but I'll read over again + with your suggestion to see how to incorporate it. The goal would be that noteworthy tags account for aliases, and noteworthy tags can also be tagged with aliases so that all of them show up under the same group. But this only applies to noteworthy tags, and maybe it would be good to allow alias groups even for things which aren't noteworthy, but could still use it; I'm not sure how many things fit into that category.

Reply:
Going to resolve this since I explicitly mention aliases on the latest version; please comment on the latest revision if you have additional feedback, though!